waku/factory/node_factory.nim

```diff
@@ -398,13 +398,13 @@ proc startNode*(
     return err("failed to start waku node: " & getCurrentExceptionMsg())
 
   # Connect to configured static nodes
-  if conf.staticnodes.len > 0:
+  if conf.relay and conf.staticnodes.len > 0:
```

Comment:
In this case I'd give some more feedback in case wrong setup:
```suggestion
  if conf.staticnodes.len > 0:
    if not conf.relay:
      return err("waku relay (--relay=true) should be set when configuring staticnodes")
```

Reply:
Ooh great idea! Thanks so much!